### ISSUE TITLE
[Tools] add APA Style Checker

### ISSUE THREAD

**Issue by keeferrourke** [ian-james/IFS#33]
This issue has been opened, as converted from a project card.

@ian-james mentioned that it would be useful to add a style-checker for written works. This should be implemented as one of the core-tools.

**Comment by keeferrourke** [comment]
After a bit of searching, I found this tool <https://github.com/JonathanAquino/apacheck>; there is a demo online at: <http://jonathanaquino.com/apacheck/>.

It seems pretty comprehensive with its ruleset. I am not sure how easy it would be to adapt it as an IFS tool, but at the very least it might serve as a reference for writing a python script (perhaps covering other styles? [stretch goal??]).

I'm not sure what kind of priority to set for this issue.

**Comment by ian-james** [comment]
That is a good find  I think I only saw one other APA tool that looked
applicable. I had hoped we would find that tool for free but perhaps not.

The APA checker was the big ask from the psych side so that might be top
priority or the skills/assignment check list.

**Comment by keeferrourke** [comment]
### Status Update

Working on writing a Python3 module that checks for most common errors in APA. The regex is blending my brain, and I haven't had a chance to test the Python yet (still writing regex for the object classes), but this is what I have so far: https://regex101.com/r/hXy2kp/1

**Comment by keeferrourke** [comment]
### Status Update
Working Python module is now located over at https://github.com/keeferrourke/pyapa

It has not been thoroughly tested, but should be mostly functional. I'm working to make it installable from pypi `pip` at the moment. An IFS-flavoured wrapper script will be in the works shortly.

**Comment by keeferrourke** [comment]
New APA Style Checker tool has been pushed. It works pretty well in my limited testing! See commit 4d1b247231bfb3a2fc24a3289d7b260a7bfed9ea

**Comment by keeferrourke** [comment]
Commit d95ab82 decouples a lot of my Python APA module from IFS. It is now necessary to install `pyapa` from https://pypi.python.org/pypi/pyapa in order to use the `apa_check.py` tool. The installation script has been updated to reflect this change.

This has been done to clean up the source a little bit, and make the APA Checker independent from IFS, in case anyone else encounters a similar problem that we did (in that we needed a tool to check APA style, and there were no viable console options).
